packages/nimble-components/src/table-column/mapping/mappings/spinner/index.ts

```diff
@@ -0,0 +1,69 @@
+import {
+    attr,
+    html,
+    observable,
+    ref,
+    ViewTemplate
+} from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { Mapping } from '..';
+import { spinnerTag } from '../../../../spinner';
+import { styles } from '../styles';
+import { template } from '../template';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-mapping-spinner': MappingSpinner;
+    }
+}
+
+/**
+ * Maps values to a spinner
+ */
+export class MappingSpinner extends Mapping {
+    @attr()
+    public label: string | null = null;
+
+    /** @internal */
+    public span: HTMLSpanElement | null = null;
+
+    /** @internal */
+    @observable
+    public isValidContentAndHasOverflow = false;
+
+    /** @internal */
+    // prettier-ignore
+    public cellViewTemplate: ViewTemplate = html<MappingSpinner>`
+        <${spinnerTag}
+            title="${x => x.label}"
+            aria-label="${x => x.label}">
+        </${spinnerTag}>`;
+
+    /** @internal */
+    // prettier-ignore
+    public groupHeaderViewTemplate: ViewTemplate = html<MappingSpinner>`
+        <${spinnerTag}
+            title="${x => x.label}"
+            aria-label="${x => x.label}">
+        </${spinnerTag}>
+        <span
+            ${ref('span')}
+            @mouseover="${x => {
+        x.isValidContentAndHasOverflow = !!x.label && x.span!.offsetWidth < x.span!.scrollWidth;
+    }}"
+            @mouseout="${x => {
+        x.isValidContentAndHasOverflow = false;
+    }}"
+            title=${x => (x.isValidContentAndHasOverflow ? x.label : null)}
+        >
+            ${x => x.label}
+        </span>`;
+}
+
+const spinnerMapping = MappingSpinner.compose({
+    baseName: 'mapping-spinner',
```

Comment:
Currently our folder structure aligns with the element name. I don't think there is a strong reason to diverge here.
We should either have:
```
/src/mapping-base
/src/mapping-icon
/src/mapping-spinner
/src/mapping-text
```

Or maybe following the table-column pattern as there might be more of these in the future:
```
/src/mapping/base
/src/mapping/icon
/src/mapping/spinner
/src/mapping/text
```

Reply:
Went for the latter option.